src/main/java/org/jahia/services/usermanager/ldap/LDAPUserGroupProvider.java

```diff
@@ -384,6 +388,19 @@ public boolean verifyPassword(String userName, String userPassword) {
                 // that needs to be removed from the user DN for the lookup to succeed.
                 ctx.lookup(LdapUtils.newLdapName(userCacheEntry.getDn()));
                 logger.debug("Password verified for {} in {} ms", userName, System.currentTimeMillis() - startTime);
+           
+                try {
+                    JCRTemplate.getInstance().doExecuteWithSystemSession((JCRSessionWrapper session) -> {
+                        final String userRelativePath = ServicesRegistry.getInstance().getJahiaUserManagerService().getUserSplittingRule().getRelativePathForUsername(userName);
+                        final String providerKey = getKey();
+                        JahiaGroupManagerService.getInstance().flushMembershipCache("/users/providers/" + providerKey + userRelativePath, session);
```

Comment:
This won't work if the provider is mounted at site level

Reply:
Indeed, PR updated with code to consider when it's mounted at site level (and tested)